go/l1client/client.go

```diff
@@ -0,0 +1,37 @@
+package l1client
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+)
+
+// EthereumClient defines the interface for RPC communications with the ethereum nodes
+// Some of these methods are composed calls that should be decoupled in the future (ie: BlocksBetween or IsBlockAncestor)
+type EthereumClient interface {
```

Comment:
@otherview Minor, but it feels odd that there's a file called `ethereum_client.go`, but the `EthereumClient` type isn't defined in it.

Reply:
yeah, the file should be named consistently

Reply:
That's the typical package naming.

This is my favorite a fruit example 😄  :

```
fruitclient/client.go (interface)
fruitclient/banana_client.go (banana implementation of the client)
fruitclient/apple_client.go (apple implementation of the client)
fruitclient/client_test.go (tests that run against the implementation)
fruitclient/banana_client_test.go (instantiates the banana impl and runs the client_test.go tests)
fruitclient/banana_client_test.go (instantiates the apple impl and runs the client_test.go tests)
```

Reply:
Resolved in chat 👍